t/re/pat.t

```diff
@@ -1902,14 +1902,27 @@ EOP
             }
         }
         {
-            fresh_perl_is('
+            local $::TODO = "this flaps on github cygwin vm, but not on cygwin iron #18129"
+              if $^O eq "cygwin";
+            my $expected = "Timeout";
+            my $code = '
                 BEGIN{require q(test.pl);}
                 watchdog(3);
-                $SIG{ALRM} = sub {print "Timeout\n"; exit(1)};
+                $SIG{ALRM} = sub {print "'.$expected.'\n"; exit(1)};
                 alarm 1;
                 $_ = "a" x 1000 . "b" x 1000 . "c" x 1000;
                 /.*a.*b.*c.*[de]/;
-            ',"Timeout",{},"Test Perl 73464")
+                print "increased the multipliers in the regex above to run the regex longer";
+            ';
+            my ($iter, $result, $status);
+            for my $i (1..50) { # this flaps on github cygwin vm, but not on cygwin iron #18129
```

Comment:
Unless I'm misunderstanding this, this will make the test 147s longer on all platforms to try and diagnose what's happening on one platform, that doesn't seem sensible.

I suggest adding the TODO without other changes as one PR, and look at how to get more value from the test separately.

(I'll also note that I don't like the test in the first place: improved optimization in the regexp engine will cause this to fail without signalling any error.)

Reply:
> Unless I'm misunderstanding this, this will make the test 147s longer on all platforms to try and diagnose what's happening on one platform, that doesn't seem sensible.
>
> I suggest adding the TODO without other changes as one PR, and look at how to get more value from the test separately.

It'd be 50 seconds (1 second alarm), and i've pushed a change to only do that on cygwin.

The value of doing it on ALL cygwin machines is to help find various machines where it does and doesn't fail to see which differences cause this.

> (I'll also note that I don't like the test in the first place: improved optimization in the regexp engine will cause this to fail without signalling any error.)

That's why i added a print after the regex that indicates completion before alarm is an error and how to fix it. :)